src/polygonfill2d.jl

```diff
@@ -28,12 +28,12 @@ end
 """
 
 function (f::FloodFill)(
-    res::AbstractArray{T,2},
-    verts::Vector{CartesianIndex{2}},
+    res::AbstractArray{T, N},
+    verts::Vector{CartesianIndex{N}},
     x::Int,
     y::Int,
-    ) where {T<:Colorant}
-    if (res[y, x] != f.current_value || res[y, x] == f.fill_value) return end
+    ) where {T<:Colorant, N}
+    if (res[y, x] != f.current_value || res[y, x] == f.fill_value) return res end
     if checkbounds(Bool, res, y, x) res[y, x] = f.fill_value end
     f(res, verts, x + 1, y)
     f(res, verts, x - 1, y)
```

Comment:
This can still be `{T, 2}` as the implementation is specified for matrix; this won't affect the method ambiguity on `draw` so no need to change.